lib/Conversion/ImportVerilog/Expressions.cpp

```diff
@@ -427,24 +427,35 @@ struct RvalueExprVisitor {
   Value visit(const slang::ast::ElementSelectExpression &expr) {
     auto type = context.convertType(*expr.type);
     auto value = context.convertRvalueExpression(expr.value());
+    if (!type || !value)
+      return {};
+    if (auto *constValue = expr.selector().constant) {
+      auto lowBit = constValue->integer().as<uint32_t>().value();
+      return builder.create<moore::ExtractOp>(loc, type, value, lowBit);
+    }
```

Comment:
Can you add a check that the `constValue` fits into 32 bits, and only pick this path if that's the case? Otherwise we can still lower to the dynamic extract. Just to make sure that we don't silently overflow or fail if somebody types `arr[123456789123456789123456789123456789]` in the SystemVerilog input :grin:

Reply:
Also, you probably want to check that `constValue->isUnknown()` is false, meaning that the value does not have any X or Z in them. Otherwise we would silently drop X or Z when converting to the `lowBit` integer value.

Reply:
> Also, you probably want to check that `constValue->isUnknown()` is false, meaning that the value does not have any X or Z in them. Otherwise we would silently drop X or Z when converting to the `lowBit` integer value.

You're totally right! I'll add this check.

Reply:
> Can you add a check that the `constValue` fits into 32 bits, and only pick this path if that's the case? Otherwise we can still lower to the dynamic extract. Just to make sure that we don't silently overflow or fail if somebody types `arr[123456789123456789123456789123456789]` in the SystemVerilog input 😁

slang will check the bit width and overflow like 
```
source.sv:2:6: warning: signed integer literal overflows 32 bits, will be truncated to -1097262461 [-Wint-overflow]
bit [123456789123:0] arr;
     ^
source.sv:2:6: error: packed type of 1097262462 bits is too large; maximum width is 16777215 bits
bit [123456789123:0] arr;
```

Reply:
Slang is always checking the `constValue` whether is `x/z`. Like
```
source.sv:6:11: error: cannot refer to element 1'bx of 'logic[3:0]' [-Windex-oob]
    b = a[1'bx];
          ^~~~
source.sv:8:14: error: value must not have any unknown bits
    b = a[3-:1'bx];
             ^~~~
source.sv:9:10: error: value must not have any unknown bits
    a[3-:1'bz] = b;
```

Reply:
Ah that's nice! Thanks a lot. I didn't realize that SV array dimensions would be limited to 32 bits by Slang. That's great though, in that case we don't really need to generate diagnostic errors in the code. Maybe just an `assert(...)` to encode our assumption about <=32bits and no X/Z in the code somewhere.

Reply:
I only add `assert()` for no X/Z. For `<= 32 bits`, I'll add later.